core/editor/src/document/document_message_handler.rs

```diff
@@ -566,33 +574,25 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					responses.push_back(operation.into());
 				}
 			}
-			ReorderSelectedLayers(delta) => {
-				let selected_layer_paths: Vec<Vec<LayerId>> = self.selected_layers_sorted();
+			MoveSelectedLayersTo { path, insert_index } => {
+				responses.push_back(DocumentMessage::CopySelectedLayers.into());
+				responses.push_back(DocumentMessage::DeleteSelectedLayers.into());
+				responses.push_back(DocumentMessage::PasteLayers { path, insert_index }.into());
+			}
+			ReorderSelectedLayers(relative_positon) => {
 				let all_layer_paths = self.all_layers_sorted();
-
-				let max_index = all_layer_paths.len() as i64 - 1;
-				let num_layers_selected = selected_layer_paths.len() as i64;
-
-				let mut selected_layer_index = -1;
-				let mut next_layer_index = -1;
-				for (i, path) in all_layer_paths.iter().enumerate() {
-					if *path == selected_layer_paths[0] {
-						selected_layer_index = i as i32;
-						// Skip past selection length when moving up
-						let offset = if delta > 0 { num_layers_selected - 1 } else { 0 };
-						next_layer_index = (selected_layer_index as i64 + delta as i64 + offset).clamp(0, max_index) as i32;
-						break;
+				if let Some(insert_path) = all_layer_paths
+					.iter()
+					.position(|path| self.layerdata(&path).selected)
```

Comment:
This gets the position of the first selected layer, but what if you're moving several layers down? Will the next line not then grab the second selected layer? Previously `offset` handled this case with contiguous selections.

Reply:
Ah yes, I had planned to select the first layer that was not part of your selection but must have forgotten in the end